firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/CRCElements.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ * Chunks.cpp
+ *
+ *  Created on: Nov 3, 2020
+ *      Author: Renji Panicker
+ *
+ * Unit tests to confirm behavior of chunk splitter
+ *
+ */
+
+#include "Pufferfish/Protocols/CRCElements.h"
+
+#include "Pufferfish/HAL/CRCChecker.h"
+#include "Pufferfish/Test/Util.h"
+#include "Pufferfish/Util/Array.h"
+#include "Pufferfish/Util/Endian.h"
+#include "catch2/catch.hpp"
+
+namespace PF = Pufferfish;
+
+SCENARIO("Protocols::CRCElement behaves correctly", "[CRCElement]") {
+  GIVEN("A CRC element") {
+    constexpr size_t buffer_size = 254UL;
+    using TestCRCElementProps = PF::Protocols::CRCElementProps<buffer_size>;
+    using TestCRCElement = PF::Protocols::CRCElement<TestCRCElementProps::PayloadBuffer>;
+    PF::HAL::SoftCRC32 crc32c{PF::HAL::crc32c_params};
+
+    WHEN("when good data is given as input") {
```

Comment:
Maybe it would be more specific to say "when the CRC and payload fields are parsed from a valid body"

Reply:
I have updated the tests along with test descriptions.